bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -1061,4 +1077,15 @@ private void setContentAssistsEnablement(boolean enable) {
 	private void updateContentAssistAvailability() {
 		setContentAssistsEnablement(findReplaceLogic.isAvailableAndActive(SearchOptions.REGEX));
 	}
+
+	private void decorate() {
+		searchBar.addModifyListener(event -> {
+			if (regexSearchButton.getSelection()) {
+				searchDecoration.decorateA(controldecoration, getFindString());
+			} else
+				controldecoration.hide();
```

Comment:
Only the `if-else` block should be in the method and you should call it twice (I saw that the same code is used in the method `createRegexSearchButton`).
Adding the `ModifyListener` is only needed once so move it out of the method and into the caller.

Reply:
done